### DISCUSSION TITLE
Looking for a AWS Cognito Strategy

### DISCUSSION THREAD

**Discussion by jeremyforan** [sergiodxa/remix-auth#308] [Q&A]
Hello,

I am looking to use AWS Cognito with remix-auth. Previous issues that have been posted seems to indicate that people have written their own, and have bumped into some problems along the way.

It would be great if there was a Cognito Strategy for OIDC / App Client connects.

**Comment by jon-karlsen** [comment]
I'm using the [remix-auth-oauth2](https://github.com/sergiodxa/remix-auth-oauth2) strategy with Cognito. It's straightforward to set up following the documentation and works well.

**Reply by cjoecker** [reply]
@jon-karlsen could you use cognito with your own UI for email and password in remix? Or with the AWS hosted UI?

**Reply by jon-karlsen** [reply]
@cjoecker We're using the AWS hosted UI.

**Reply by cjoecker** [reply]
I think is worth making a strategy for login with own UI since the hosted UI has a lot of limitations. Amplify v6 is supporting only NextJS and the only way of doing it is v5, will not be maintained soon.

**Comment by cjoecker** [comment]
I commented [here](https://github.com/aws-amplify/amplify-js/issues/12660#issuecomment-2760604618) how to use Cognito now with Remix / React Router v7.
